src/component/Slider/Slider.jsx

```diff
@@ -0,0 +1,275 @@
+
+    if (isScrolling) {
+      const scrX = ((clientX - currenClientX )/carouselRef.current.clientWidth) ;
+      setScrollingState({
+        ...scrollingState,
+        scrollX: scrX
+      });
+    }
+  }
+
+  const onTouchMove = e => {
+    let currenClientX = e.touches[0].clientX;
+    handleSlideMove(currenClientX);
+    }
+
+  const onMouseMove = e => {
+    e.preventDefault();
+    let currenClientX = e.clientX;
+    handleSlideMove(currenClientX);
+  }
+
+  const changeSlideHandle = () => {
+    if (scrollingState.scrollX > 0.3) {
+      if (currentIndex === (length - slidesPerView) && !infiniteLoop) {
+        currentSlideShow();
+        setScrollingState({
+          ...scrollingState,
+          isTransition: true,
+          isScrolling: false,
+          scrollX: 0,
+          clientX: 0,
+        })
+        return;
+      }
+
+      onClickNext();
+    } else if (scrollingState.scrollX < -0.3) {
+      onClickPrev();
+    } else {
+      currentSlideShow();
+    }
+    setScrollingState({
+      ...scrollingState,
+      isTransition: true,
+      isScrolling: false,
+      scrollX: 0,
+      clientX: 0,
+    })
+  }
+
+  const currentSlideShow = () => {
+    setScrollingState({
+      ...scrollingState,
+      isTransition: true,
+      isScrolling: false,
+      scrollX: 0,
+    })
+  }
+
+  const onMouseUp = () => {
+    if (!scrollingState.isScrolling) {
+      return;
+    }
+    changeSlideHandle();
+  }
+
+  const onTouchEnd = () => {
+    if (!scrollingState.isScrolling) {
+      return;
+    }
+
+    changeSlideHandle();
+  }
+
+  const onMouseLeave = e => {
+    if (!scrollingState.isScrolling) {
+      return;
+    }
+
+    e.preventDefault();
+    changeSlideHandle();
+  }
+
+  const onTouchCancel = () => {
+    if (!scrollingState.isScrolling) {
+      return;
+    }
+
+    changeSlideHandle();
+  }
+
+  const renderPrevSlide = () => {
+    const output = [];
+    for (let i = 0; i < slidesPerView; i += 1) {
+      output.push(children[length - 1 - i]);
+    }
+
+    return output.reverse();
+  };
+
+  const renderNextSlide = () => {
+    const output = [];
+    for (let i = 0; i < slidesPerView; i += 1) {
+      output.push(children[i]);
+    }
+
+    return output;
+  };
+
+  return (
+    <div 
+      className="carousel__container"
+      >
+      <div
+        ref={carouselRef}
+        className={`carousel__slides ${scrollingState.isTransition ? 'transition' : 'transition-stop'}`}
+        style={{
+          transform: `translateX(-${(currentIndex + scrollingState.scrollX) * 100}%)`,
+          width: `calc(100%/${slidesPerView})`,
+        }}
+        onTouchStart={onTouchStart}
+        onTouchMove={onTouchMove}
+        onTouchEnd={onTouchEnd}
+        onTouchCancel={onTouchCancel}
+        onMouseDown={onMouseDown}
+        onMouseUp={onMouseUp}
+        onMouseMove={onMouseMove}
+        onMouseLeave={onMouseLeave}
+        onTransitionEnd={handleTransitionEnd}
```

Comment:
* inconsistent handlers names
* a pattern `on<Something>` describe an event. `onMouseDown`, `onTouchEnd` etc. A handler is a function which has only one purpose - handle an event. It's a method. A method name should be described as an action description. For example I see an event `onMouseLeave`. To handle it I have to define method with name `handleMouseLeave`

Reply:
I renamed all the functions by the pattern you suggested.